src/swupd_lib/curl.c

```diff
@@ -457,7 +457,9 @@ enum download_status process_curl_error_codes(int curl_ret, CURL *curl_handle)
 		}
 	} else { /* download failed but let our caller do it */
 		long response = 0;
-		curl_easy_getinfo(curl_handle, CURLINFO_RESPONSE_CODE, &response);
+		if (curl_easy_getinfo(curl_handle, CURLINFO_RESPONSE_CODE, &response) != CURLE_OK) {
+			response = -1; // Force error
```

Comment:
I wonder if we should go ahead and decode the error code with `curl_easy_strerror` to produce an informative message here.
https://curl.se/libcurl/c/libcurl-errors.html#CURLEOK